### DISCUSSION TITLE
negative values using clamp

### DISCUSSION THREAD

**Discussion by stereonom** [tailwindlabs/tailwindcss#9236] [Ideas]
Hi.

in my project I use some special-values for my spacing:
```
spacing: {
	'foo': 'clamp(1.143rem, 4vw - 1.25rem, 2.857rem)'
},
```

When I use it like this: `mb-foo` tailwind correctly generates my class and adds the clamp-value as `margin-top`. No surprise.

When I use it as a negative value `-mt-foo` no class gets generated.
I think this is, because you can't just add a `-` before the clamp-value and call it a day.
But what could get generated is:
```
.-mt-foo {
	margin-top: calc( clamp(1.143rem, 4vw - 1.25rem, 2.857rem) * -1);
}
```

I just wanted to leave this here for discussion. Perhaps in the future, when more `clamp` is used this would be a cool feature :-)

**Comment by thecrypticace** [comment]
This is an interesting idea. I think there could be a good bit of complexity in trying to support a feature like that. However, we do something fairly similar for calc & var. We're a bit focused on other things at the moment but this might be a simple enough addition.

**Comment by thecrypticace** [comment]
Yeah I figured I'd use a short time on my lunch to see how easy this was. Looks like… very! https://github.com/tailwindlabs/tailwindcss/pull/9237 — it'll be in our next release :)

**Reply by stereonom** [reply]
😳 Wow. Just wow. Thanks. 💙
